### PR TITLE
Fix more leaks detected by ASAN

### PR DESCRIPTION
**Pull request by dioni21**
### Motivation and Context

Found some Leaks using ASAN compilation

### Description

This commit fixes a bunch of missing free() calls created in a10d50f99

### How Has This Been Tested?

Build and run.  Message is gone.

### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [X] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by dioni21** [comment]
@behlendorf could you please take a look at those test failures?  I thought these changes were extremely simple, and the most intriguing is that only Ubuntu 18.04 failed.

File: lib/libzfs/libzfs_dataset.c [hunk 1]
```diff
@@ -931,8 +931,13 @@ libzfs_mnttab_add(libzfs_handle_t *hdl, const char *special,
...
 		 */
-		if (avl_find(&hdl->libzfs_mnttab_cache, mtn, NULL) != NULL)
+		if (avl_find(&hdl->libzfs_mnttab_cache, mtn, NULL) != NULL) {
+			free(mtn->mtn_mt.mnt_special);
+			free(mtn->mtn_mt.mnt_mountp);
+			free(mtn->mtn_mt.mnt_fstype);
+			free(mtn->mtn_mt.mnt_mntopts);
 			free(mtn);
+		}
 		else
```
Comment (behlendorf) [inline lib/libzfs/libzfs_dataset.c:941]: nit: can you update the `else` block to use this style:

```c
if () {
       ....
} else {
        avl_add(&hdl->libzfs_mnttab_cache, mtn);
}
```

Comment (dioni21) [reply lib/libzfs/libzfs_dataset.c]: Done.  Sorry for that.

**Comment by dioni21** [comment]
Good addition that STYLE check, BTW!

**Pull request merged**